### PR TITLE
Drop hashicorp lru - switch to the lru impl in k8s.io/utils

### PR DESCRIPTION
**Pull request by dims**
Kubernetes has a lot of references to hashicorp's golang-lru package even after https://github.com/kubernetes/kubernetes/pull/95472 landed:
https://cs.k8s.io/?q=-lru%22&i=nope&files=&excludeFiles=&repos=kubernetes/kubernetes

Given that the golang-lru package came from:
https://github.com/golang/groupcache/tree/master/lru

We could just as well depend on that groupcache directly, however there is a catch, the hashicorp's golang-lru impl is thread safe, so essentially in this PR, we pick up LRU.go from groupcache and add a lock around it. this was added to `k8s.io/utils`:
https://github.com/kubernetes/utils/pull/209

This will help us get rid of the last of the `Mozilla Public License` dependency from kubernetes LICENSES 

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: vendor/k8s.io/utils/internal/third_party/forked/golang/LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2012 The Go Authors. All rights reserved.
```
Comment (liggitt) [inline vendor/k8s.io/utils/internal/third_party/forked/golang/LICENSE:1]: I'm +1 for picking this, but it seems off that this LICENSE doesn't get pulled into the k/k licenses in any way... should k8s.io/utils update/include this somehow?

Comment (dims) [reply vendor/k8s.io/utils/internal/third_party/forked/golang/LICENSE:1]: @liggitt good point. just realized we don't do `third_party/forked/golang/LICENSE` in kubernetes/kubernetes either. will tackle both of them together. let's get this in and i'll file an issue/pr afterwards

Comment (dims) [reply vendor/k8s.io/utils/internal/third_party/forked/golang/LICENSE:1]: https://github.com/kubernetes/kubernetes/issues/103551

Comment (dims) [reply vendor/k8s.io/utils/internal/third_party/forked/golang/LICENSE:1]: PR is here : https://github.com/kubernetes/kubernetes/pull/103556

**Comment by nikhita** [comment]
Re-adding lgtm

/lgtm
/retest

**Pull request merged**